File: 746 - Min Cost Climbing Stairs.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int dp[3] = {0};
        int n = cost.size();
        for (int i = 2; i < n + 1; i++) {
            dp[2] = min(dp[0] + cost[i - 2], dp[1] + cost[i - 1]);
            dp[0] = dp[1];
            dp[1] = dp[2];
        }
        return dp[2];
    }
};

File: 746 - Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: list[int]) -> int:
        n = len(cost)
        dp = [0] * 3
        for i in range(2, n + 1):
            dp[2] = min(dp[0] + cost[i-2], dp[1] + cost[i-1])
            dp[0], dp[1] = dp[1], dp[2]
            # print(dp)
        return dp[2]



s = Solution()
a = s.minCostClimbingStairs([0,1,2,2])
print(a)